sentry-android-core/src/main/java/io/sentry/android/core/SentryLogcatAdapter.java

```diff
@@ -0,0 +1,109 @@
+package io.sentry.android.core;
+
+import android.util.Log;
+import io.sentry.Breadcrumb;
+import io.sentry.Sentry;
+import io.sentry.SentryLevel;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * This class replaces {@link android.util.Log} with its own implementations which creates a {@link
+ * io.sentry.Breadcrumb} for each log. It only replaces log functions that meet a minimum level set
+ * by the user on the Sentry Android Gradle Plugin.
+ */
+@ApiStatus.Internal
+public final class SentryLogcatAdapter {
+
+  private static void addAsBreadcrumb(@NotNull String tag, @NotNull SentryLevel level, String msg) {
+    addAsBreadcrumb(tag, level, msg, null);
+  }
+
+  private static void addAsBreadcrumb(
+      @NotNull String tag, @NotNull SentryLevel level, Throwable tr) {
+    addAsBreadcrumb(tag, level, null, tr);
+  }
+
+  private static void addAsBreadcrumb(
+      @NotNull String tag, @NotNull SentryLevel level, String msg, Throwable tr) {
+    Breadcrumb breadcrumb = new Breadcrumb();
+    breadcrumb.setCategory("log");
+    breadcrumb.setMessage(msg);
+    breadcrumb.setLevel(level);
+    breadcrumb.setData("tag", tag);
+    if (tr != null) {
+      breadcrumb.setData("throwable", tr.getMessage());
+    }
+    Sentry.addBreadcrumb(breadcrumb);
+  }
+
+  public static int v(String tag, @NotNull String msg) {
```

Comment:
I think it probably makes sense to have `tag`, `msg` and `throwable` marked as `@Nullable` everwhere
<img width="616" alt="image" src="https://user-images.githubusercontent.com/1411808/227449250-d1d685c4-043c-4c3e-8e8c-a9cc6e9d8cba.png">